src/image_png.cpp

```diff
@@ -187,16 +187,19 @@ static void ReadGrayData(
 		png_read_row(png_ptr, dst, NULL);
 	}
 
-	// Black pixels are transparent
 	if (transparent) {
-		uint8_t ck1[4] = {0, 0, 0, 255};
-		uint8_t ck2[4] = {0, 0, 0,   0};
-		uint32_t srckey = *(uint32_t*)ck1;
-		uint32_t dstkey = *(uint32_t*)ck2;
 		uint32_t* p = (uint32_t*) pixels;
-		for (unsigned i = 0; i < w * h; i++, p++)
-			if (*p == srckey)
-				*p = dstkey;
+
+		for (unsigned i = 0; i < w * h; i++, p++) {
+			uint8_t a = (*p & 0xff000000) >> 24;
+			uint8_t b = (*p & 0x00ff0000) >> 16;
+			uint8_t g = (*p & 0x0000ff00) >> 8;
+			uint8_t r = (*p & 0x000000ff);
+
+			// Black pixels are transparent
+			if (b == 0 && g == 0 && r == 0 && a == 255)
+				*p = 0;
+		}
 	}
```

Comment:
Finally tested it under the Wii: Does indeed not work. Colors are completely wrong when transparency is used. All other graphics look fine.